osu.Game/Beatmaps/Formats/LegacyBeatmapEncoder.cs

```diff
@@ -230,10 +231,10 @@ void outputControlPointAt(double time, bool isTimingPoint)
                 LegacyEffectFlags effectFlags = LegacyEffectFlags.None;
                 if (effectPoint.KiaiMode)
                     effectFlags |= LegacyEffectFlags.Kiai;
-                if (effectPoint.OmitFirstBarLine)
+                if (timingPoint.OmitFirstBarLine)
```

Comment:
Of note, this is a subtler-than-apparent change which makes it so that if the problematic beatmap mentioned in the issue is re-exported using lazer's editor, it will auto-fix itself, as in the problematic inherited point will have the omit bar line flag set after export. Which is neat, but also technically breaks import-export stability. But I think that's fine.